File: src/main/java/Problem0029.java
import java.math.BigInteger;
import java.util.ArrayList;
import java.util.LinkedHashSet;

public class Problem0029 {

    public Problem0029() {
        long startTime = System.nanoTime();
        long result = 0;

        // Solution:
        LinkedHashSet<BigInteger> set = new LinkedHashSet<>();
        for (int i = 2; i <= 100; i++) {
            for (int j = 2; j <= 100; j++) {
                set.add(new BigInteger(String.valueOf(j)).pow(i));
            }
        }
        ArrayList<BigInteger> list = new ArrayList<>(set);
        result = list.size();

        long timeToResolve = System.nanoTime() - startTime;
        System.out.println("Result:\t" + result + "\tTime:\t" + (((double) timeToResolve / 1_000_000) > 1000 ?
                (((double) timeToResolve / 1_000_000_000) + "s") :
                (((double) timeToResolve / 1_000_000) + "ms")));
    }
}

File: src/main/cs/Problem0029.cs
﻿using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Numerics;
using Euler.test.cs;

namespace Euler.main.cs
{
    public class Problem0029
    {
        public Problem0029()
        {
            Stopwatch stopWatch = new Stopwatch();
            stopWatch.Start();
            long result = 0;

            // Solution:
            HashSet<BigInteger> set = new HashSet<BigInteger>();
            for (int i = 2; i <= 100; i++)
            for (int j = 2; j <= 100; j++)
                set.Add(BigInteger.Pow(j, i));

            result = set.Count;


            stopWatch.Stop();
            string elapsedTime = stopWatch.Elapsed.ToString();
            Console.WriteLine("Result:\t" + result + "\tTime:\t" +
                              (double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1, 2)) >= 1
                                  ? double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1)) + " s"
                                  : double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000 +
                                    " ms"));
            if (Test.DoBenchmark)
                Benchmark.AddTime(29, double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000);
        }
    }
}